controller/internal/enforcer/utils/rpcwrapper/rpc_handle.go

```diff
@@ -61,6 +59,11 @@ const (
 // NewRPCClient exported
 func (r *RPCWrapper) NewRPCClient(contextID string, channel string, sharedsecret string) error {
 
+	// If a client exists for this context kill it.
+	if _, err := r.rpcClientMap.Get(contextID); err == nil {
+		r.DestroyRPCClient(contextID)
+	}
+
```

Comment:
If a client exists should we return an error here because most likely we have a remote enforcer running which we have forgotten about. 
we need to kill that before we try to launch a new one

Reply:
I added a delete in the collectChildExitStatus, but I think there is still a possible race condition. This code should never be reached.